lib/ret_web/channels/hub_channel.ex

```diff
@@ -476,13 +489,21 @@ defmodule RetWeb.HubChannel do
   end
 
   def handle_in("block" = event, %{"session_id" => session_id} = payload, socket) do
-    socket = socket |> assign(:blocked_session_ids, socket.assigns.blocked_session_ids |> Map.put(session_id, true))
+    socket =
+      socket
+      |> assign(:blocked_session_ids, socket.assigns.blocked_session_ids |> Map.put(session_id, true))
+      |> assign(:has_blocks, true)
+
     broadcast_from!(socket, event, payload |> payload_with_from(socket))
     {:noreply, socket}
   end
 
   def handle_in("unblock" = event, %{"session_id" => session_id} = payload, socket) do
-    socket = socket |> assign(:blocked_session_ids, socket.assigns.blocked_session_ids |> Map.delete(session_id))
+    socket =
+      socket
+      |> assign(:blocked_session_ids, socket.assigns.blocked_session_ids |> Map.delete(session_id))
+      |> assign(:has_blocks, true)
```

Comment:
Why are we setting `has_blocks` to true on `unblock`? Same for line 609 below. Shouldn't we instead set it to false if `blocked_session_ids` is empty?

Reply:
nice catch. bug on my part.

Reply:
PR updated. we need to actually set it properly when either map is non-empty, across all calls